int-set/src/lib.rs

```diff
@@ -0,0 +1,165 @@
+mod bitpage;
+mod bitset;
+
+use bitset::BitSet;
+use std::ops::RangeInclusive;
+
+/// A fast & efficient unsigned integer (u32) bit set which is invertible.
+#[derive(Clone, Debug)]
+pub enum IntSet<T> {
+    /// Records a set of integers which are members of the set.
+    Standard(BitSet<T>),
+
+    /// Records the set of integers which are not members of the set.
+    Inverted(BitSet<T>),
+}
+
+impl<T: Into<u32> + Copy + Default> IntSet<T> {
+    /// Create a new empty set.
+    pub fn default() -> IntSet<T> {
+        IntSet::Standard(BitSet::<T>::default())
+    }
+
+    /// Create a new set which contains all integers.
+    pub fn all() -> IntSet<T> {
+        IntSet::Inverted(BitSet::<T>::default())
+    }
+
+    /// Return the inverted version of this set.
+    pub fn inverted(self) -> IntSet<T> {
+        match self {
+            IntSet::<T>::Standard(s) => IntSet::<T>::Inverted(s),
+            IntSet::<T>::Inverted(s) => IntSet::<T>::Standard(s),
+        }
+    }
+
+    /// Return a new version of this set with all members removed.
+    pub fn clear(mut self) -> IntSet<T> {
+        self.clear_internal_set();
+        match self {
+            IntSet::<T>::Standard(s) => IntSet::<T>::Standard(s),
+            IntSet::<T>::Inverted(s) => IntSet::<T>::Standard(s),
+        }
+    }
+
+    /// Add val as a member of this set.
```

Comment:
```suggestion
    /// Adds a value to the set.
    ///
    /// Returns `true` if the value was newly inserted.
```

(I'm just copying/adapting the docs from `HashSet::insert`)